ci-operator/step-registry/baremetalds/assisted/operator/bmh/baremetalds-assisted-operator-bmh-commands.sh

```diff
@@ -0,0 +1,197 @@
+    - infraenv_name: "{{ lookup('env', 'ASSISTED_INFRAENV_NAME') }}"
+    - pull_secret_name: "{{ lookup('env', 'ASSISTED_PULLSECRET_NAME') }}"
+    - ssh_private_key_name: "{{ lookup('env', 'ASSISTED_PRIVATEKEY_NAME') }}"
+    - ssh_public_key: "{{ lookup('file', '/root/.ssh/id_rsa.pub') }}"
+
+  tasks:
+  - name: write the cluster image set crd
+    template:
+      src: "clusterImageSet.j2"
+      dest: "clusterImageSet.yaml"
+
+  - name: write the infraEnv crd
+    template:
+      src: "infraEnv.j2"
+      dest: "infraEnv.yaml"
+
+  - name: write the clusterDeployment crd
+    template:
+      src: "clusterDeployment.j2"
+      dest: "clusterDeployment.yaml"
+EOCR
+
+cat > clusterImageSet.j2 <<EOCR
+apiVersion: hive.openshift.io/v1
+kind: ClusterImageSet
+metadata:
+  name: '{{ cluster_image_set_name }}'
+  namespace: {{ assisted_namespace }}
+spec:
+  releaseImage: '{{ cluster_release_image }}'
+EOCR
+
+cat > infraEnv.j2 <<EOCR
+apiVersion: agent-install.openshift.io/v1beta1
+kind: InfraEnv
+metadata:
+  name: {{ infraenv_name }}
+spec:
+  clusterRef:
+    name: {{ cluster_name }}
+    namespace: {{ assisted_namespace }}
+  agentLabelSelector:
+    matchLabels:
+      bla: aaa
+  pullSecretRef:
+    name: {{ pull_secret_name }}
+  proxy:
+    httpProxy: http://11.11.11.33
+    httpsProxy: http://22.22.22.55
+  sshAuthorizedKey: '{{ ssh_public_key }}'
+  ignitionConfigOverride: '{"ignition": {"version": "3.1.0"}, "storage": {"files": [{"path": "/etc/someconfig", "contents": {"source": "data:text/plain;base64,aGVscGltdHJhcHBlZGluYXN3YWdnZXJzcGVj"}}]}}'
+  nmStateConfigLabelSelector:
+    matchLabels:
+      some-user-defined-label-name: some-user-defined-label-value
+EOCR
+
+cat > clusterDeployment.j2 <<EOCR
+apiVersion: hive.openshift.io/v1
+kind: ClusterDeployment
+metadata:
+  name: {{ cluster_deployment_name }}
+  namespace: {{ assisted_namespace }}
+spec:
+  baseDomain: redhat.com
+  clusterName: {{ cluster_name }}
+  platform:
+    agentBareMetal:
+      apiVIP: ""
+      ingressVIP: ""
+      agentSelector:
+        matchLabels:
+          bla: aaa
+  provisioning:
+    imageSetRef:
+      name: '{{ cluster_image_set_name }}'
+    sshPrivateKeySecretRef:
+      name: {{ ssh_private_key_name }}
+    installStrategy:
+      agent:
+        sshPublicKey: '{{ ssh_public_key }}'
+        networking:
+          clusterNetwork:
+            - cidr: 10.128.0.0/14
+              hostPrefix: 23
+          serviceNetwork:
+            - 172.30.0.0/16
+        provisionRequirements:
+          controlPlaneAgents: 1
+  pullSecretRef:
+    name: {{ pull_secret_name }}
+EOCR
+
+echo "Running Ansible playbook to create kubernetes configs"
+ansible-playbook assisted-installer-crds-playbook.yaml
+
+echo "Assisted Installer pods"
+oc get pods -n assisted-installer
+
+echo "Metal3 pods"
+oc get pods -A | grep metal3
+
+echo "VM status"
+virsh list --all
+vm=""
+vm=$(virsh list --all | grep "shut off" | awk '{print $2}')
```

Comment:
same as ``virsh list --inactive``?

Reply:
It's really close, but this gives just the name of the vm without the header. It would be something needed to refactor if there is more than one offline VM.

Reply:
ack